client/blocks/signup-form/index.jsx

```diff
@@ -1245,6 +1260,9 @@ class SignupForm extends Component {
 						queryArgs={ this.props.queryArgs }
 						userEmail={ this.getEmailValue() }
 						{ ...gravatarProps }
+						labelText={ this.props.translate( 'Your email address' ) }
```

Comment:
I think this would set the label text to that value for everyone using the passwordless signup (not the Social First as it is using PasswordlessSignupForm in another component). 
Anyway, since we're shifting to being social first, this is not a big of a problem. So see if you can set this value from the AcceptInvite, but if it is hard to do then nevermind 👍